data-prepper-core/src/test/java/com/amazon/dataprepper/server/DataPrepperServerTest.java

```diff
@@ -0,0 +1,78 @@
+package com.amazon.dataprepper.server;
+
+import com.amazon.dataprepper.pipeline.server.DataPrepperServer;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class DataPrepperServerTest {
+
+    private void setRegistry(PrometheusMeterRegistry prometheusMeterRegistry) {
+        Metrics.globalRegistry.getRegistries().iterator().forEachRemaining(meterRegistry -> Metrics.globalRegistry.remove(meterRegistry));
+        Metrics.addRegistry(prometheusMeterRegistry);
+    }
+
+    @Test
+    public void testGetMetrics() throws IOException, InterruptedException, URISyntaxException {
+        final String scrape = UUID.randomUUID().toString();
+        final PrometheusMeterRegistry prometheusMeterRegistry = new PrometheusRegistryMockScrape(PrometheusConfig.DEFAULT, scrape);
+        setRegistry(prometheusMeterRegistry);
+        final DataPrepperServer dataPrepperServer = new DataPrepperServer(8080);
+        dataPrepperServer.start();
+
+        HttpRequest request = HttpRequest.newBuilder(new URI("http://127.0.0.1:8080/metrics/prometheus"))
+                .GET().build();
+        HttpResponse response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        Assert.assertEquals(200, response.statusCode());
+        Assert.assertEquals(scrape, response.body());
+        dataPrepperServer.stop();
+    }
+
+    @Test
+    public void testScrapeFailure() throws IOException, InterruptedException, URISyntaxException {
+        setRegistry(new PrometheusRegistryThrowingScrape(PrometheusConfig.DEFAULT));
+        final DataPrepperServer dataPrepperServer = new DataPrepperServer(8080);
+        dataPrepperServer.start();
+
+        HttpRequest request = HttpRequest.newBuilder(new URI("http://127.0.0.1:8080/metrics/prometheus"))
+                .GET().build();
+        HttpResponse response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        Assert.assertEquals(500, response.statusCode());
+        dataPrepperServer.stop();
+    }
+
+    private static class PrometheusRegistryMockScrape extends PrometheusMeterRegistry {
```

Comment:
Just a note: Used this instead of mocking because micrometer does many function calls on the meter registry when its added to the global registry, and we'd have to mock dozens of methods to not get NPEs.

Reply:
:+1: was going to ask about that